packages/cozy-mespapiers-lib/src/components/Search/search.js

```diff
@@ -78,43 +72,47 @@ export const makeContactTags = contact => {
   return contactTags
 }
 
-export const addFileDoc = (index, doc, scannerT) => {
-  return index.add({
-    ...doc,
-    flexsearchProps: {
-      tag: makeFileTags(doc),
-      name: doc.name,
-      translatedQualificationLabel: scannerT(
-        `items.${doc.metadata.qualification.label}`
-      )
-    }
-  })
+export const addFileDoc = (index, doc, t) => {
+  if (hasQualifications(doc)) {
+    return index.add({
+      ...doc,
+      _type: doc._type || doc.type || FILES_DOCTYPE, // should be useless after solving https://github.com/cozy/cozy-libs/issues/2023
+      flexsearchProps: {
+        tag: makeFileTags(doc),
+        translatedQualificationLabel: t(
+          `items.${doc.metadata.qualification.label}`
+        )
+      }
+    })
+  }
 }
 
 export const addContactDoc = (index, doc) => {
   return index.add({
     ...doc,
+    _type: doc._type || doc.type || CONTACTS_DOCTYPE, // should be useless after solving https://github.com/cozy/cozy-libs/issues/2023
     flexsearchProps: {
       tag: makeContactTags(doc),
-      fullname: doc.fullname,
-      birthday: doc.birthday,
-      birthcity: doc.birthcity,
-      email: doc.email?.[0]?.address,
-      civility: doc.civility,
-      company: doc.company,
-      jobTitle: doc.jobTitle
+      email: doc.email?.[0]?.address
```

Comment:
Why putting email here and not just declare it in the index like the other fields ?

Reply:
oui car c'est un tableau d'object, et l'algo final de recherche sur l'email est plus compliqué que ça. A ce stade, je préfère simplifier, et isoler l'email dans une string, et ne prenant que l'adresse du premier email.

Reply:
ça devrait pouvori se déclarer comme ça dans l'index pour avoir toutes les adresses mails : `email[]:address`.
Cf https://github.com/nextapps-de/flexsearch#complex-documents

Reply:
modifié dans le dernier commit